### DISCUSSION TITLE
(Solution) VPN new install, client connects but no network or internet access

### DISCUSSION THREAD

**Discussion by MatejsLab** [pivpn/pivpn#1565] [Support] [answered]
Hi 
I have issue with newlly installed PiVPN, as client connects via TCP to VPN, but no network or internet access can be established on the client.  PiVPN running on Pi4 with clasic 32 bit OS. 

In the past i couldn't connect via UDP at all because of ISP, but PiVPN over TCP worked fine for the last few years, it is slow thoo at 3Mbps, despite that we have 100Mbps internet.

I wass seeking for faster VPN conection and everyone sugested UDP is faster... but i cant install new VPN that would route traffic properly(out of the box), i also tried installing clasic OpenVPN on Ubuntu and I faced exactly the same issue... TCP or UDP both fail to provide network access or internet access, on client when conected to VPN.

I do have a backup to recover to the last working version thoo, as now i have no working VPN.
I am not really a VPN guru so i don't know yet what to tweak or where tho check how VPN fails to rute traffic.

It would be awesome to get access to the previous stable PiVPN version that is working normally thoo as i guess it would help a lot(as restoring old PI image is daunthing as it needs updates), I tried stable and testing version and both misbehave the same.

Loking forward to 100Mbps VPN connection to home someday probably with some more serious software than openvpn. Any sugestions?

I appriciate the PiVPN project thoo, so i guess there is a stable version in some archive, i just dont know how or where to find the right stable version? 

Thanks in advance,
Matej

**Comment by MatejsLab** [comment, marked as answer]
Actually i found a solution for PiVPN running OpenVPN, by runing **_pivpn -d_** (debug comand), which found mising ip tables config (masquarade), that was probably not set by the installer, but it should be i guess.

Thankfully that debug comand asks you to fix it by itself just by hitting **y** (yes).

I did not know for that solution but Chris at CrossTalk Solutions gave me a hint in [this PiVPN install video ](https://youtu.be/Q4zlrc0F4NU), actually he was installing Wireguard but he faced a similar issue that he resolved by **pivpn -d**.

Now with the fresh stable install of  PiVPN running OpenVPN with ip tables fixed... i get better speed via TCP, about 25Mbps down and 4 to 15Mbps Up which is great improvement.

I also tried Wireguard for experiment, but i still have no luck with UDP connection through my ISP, it seems Wireguard is UDP only.

**Comment by coolapso** [comment]
Moving to discussions: https://github.com/pivpn/pivpn#feedback--support
